### DISCUSSION TITLE
Mixed models on the same collection for Mongo

### DISCUSSION THREAD

**Discussion by Xqua** [prisma/prisma#23738] [Q&A]
### Question

I would like to know if it could be in the plans to add support for a particular feature for Prisma with Mongo DB (though this could easily apply to SQL based DB as wells).

I feel like calling it Meta Models ? 

Le'ts assume we have a collection `jobs`, I can have many types of jobs, and maybe I have a field type enum or string or whatever differentiator. 

For example:
```prisma
model Job {
    id                String   @id @default(auto()) @map("_id") @db.ObjectId
    type              String
    createdAt         DateTime @default(now())
    updatedAt         DateTime @default(now())
    finishedAt        DateTime @default(now())

    // All jobs
    status            String  @default("CREATED")
    progress          Float   @default(0.0)
    input             Json?
    output            Json?

    @@map("jobs")
}
```
now let's say I have three types of jobs: Type1Job Type2Job and Type3Job and they need some extra fields like: 
```
model Type1Job {
    inherit from Job
    filteredWith type == "type1"
    extrafield    String[]

    @@map("jobs")
}
```

etc 

It would be amazing to be able to do this without creating many collections, or end up having very long models with most of the fields ending up having to be optional because its only shared by some. 

Anyway to do this or that it could be implemented in the future ? 

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment]
Hi @Xqua 👋

Thank you for your question.

Currently, Prisma does not natively support model inheritance. Each model in Prisma corresponds to a table in SQL databases or a collection in MongoDB, and there's no built-in mechanism to inherit fields from one model to another. However, there is an open [issue](https://github.com/prisma/prisma/issues/12604) for adding support for model inheritance in Prisma. You can contribute to the discussion and add your use case to help prioritize this feature.

While Prisma doesn't support model inheritance natively, there are third-party solutions like Zenstack that add an extra layer of features to the Prisma schema, including model inheritance. You can learn more about this in their [blog post](https://www.prisma.io/docs/orm/prisma-schema/data-model/table-inheritance#third-party-solutions) about polymorphism in Prisma ORM.

In the meantime, you can use single table inheritance (STI) or multi-table inheritance (MTI) as a workaround. These methods involve sharing properties across multiple models. You can learn more about these methods in the [Prisma documentation](https://www.prisma.io/docs/orm/prisma-schema/data-model/table-inheritance#sharing-properties-across-multiple-models).

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
